falcon/request.py

```diff
@@ -1739,9 +1741,19 @@ def get_param_as_json(self, name, required=False, store=None, default=None):
         if param_value is None:
             return default
 
+        handler = self.options.media_handlers.find_by_media_type(
+            MEDIA_JSON, MEDIA_JSON, raise_not_found=False
+        )
+        if handler is None:
+            handler = _DEFAULT_JSON_HANDLER
+
         try:
-            val = json.loads(param_value)
-        except ValueError:
+            # TODO(CaselIT): find a way to avoid encode + BytesIO if handlers
```

Comment:
Yes, this will ding performance, but hopefully it isn't used often enough and the overhead is not great enough to pose a serious problem. But maybe down the road if we unity handling interfaces with the WS handlers by including a deserialize from `bytes` this can be resolved.

Reply:
yes, that's an option. I guess this is one thing to consider for v4.

Reply:
I'll try to channel my thoughts into a proposal for 3.x or 4.0: https://github.com/falconry/falcon/pull/1768#issuecomment-738896304